src/main/java/com/exactpro/th2/sailfish/utils/CompareFilter.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright 2021-2021 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.exactpro.th2.sailfish.utils;
+
+import java.math.BigDecimal;
+import java.text.DecimalFormatSymbols;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.Temporal;
+import java.util.Objects;
+
+import com.exactpro.sf.aml.scriptutil.ExpressionResult;
+import com.exactpro.th2.common.grpc.FilterOperation;
+
+public class CompareFilter implements IOperationFilter {
+
+    private final Object value;
```

Comment:
Can be replaced with more certain type
```suggestion
    private final Comparable<?> value;
```